src/Particle.js

```diff
@@ -23,8 +23,40 @@ class Particle {
 	 * @param  {Object} options Options to be used for all requests (see [Defaults](../src/Defaults.js))
 	 */
 	constructor(options = {}) {
+		// todo - this seems a bit dangerous - would be better to put all options/context in a contained object
 		Object.assign(this, Defaults, options);
+		this.context = {};
 		this.agent = new Agent(this.baseUrl);
+		this.setContext({name:'tool', value: this.tool});
```

Comment:
Where does `this.tool` come from? It's not in `Defaults` and is not documented in the constructor documentation. How would people know to add this argument?